descqa/CorrelationsTwoPoint.py

```diff
@@ -0,0 +1,324 @@
+                #RemovePixelFactor = np.std(original_pixels_count)
+                #original_pixels = original_pixels[original_pixels_count > original_pixels_count.min() + RemovePixelFactor]
+                #There is a uncertainity in the number 200
+                original_pixels = original_pixels[original_pixels_count > self.RemovePixels]
+                if (ramin <=100) & (ramax >= 260):
+                    ra = np.where(ra <= 100, ra+360, ra)
+                    ramin = ra.min()
+                    ramax = ra.max()
+                    rand_ra = np.random.uniform(ramin, ramax, randomN)
+                    rand_ra = np.where(rand_ra >= 360, rand_ra - 360, rand_ra)
+                else:
+                    rand_ra = np.random.uniform(low=ramin, high=ramax, size=randomN)
+                rand_sindec = np.random.uniform(low=np.sin(decmin * np.pi / 180.), high=np.sin(decmax * np.pi / 180.), size=randomN)
+                rand_dec = np.arcsin(rand_sindec) * 180. /np.pi
+                rand_pixels = self.return_healpixel(rand_ra, rand_dec, nside)
+                upixels = np.in1d(rand_pixels, original_pixels)
+                rand_ra = rand_ra[upixels]
+                rand_dec = rand_dec[upixels]
+
+                rand_cat = treecorr.Catalog(ra=rand_ra, dec=rand_dec, ra_units='degrees', dec_units='degrees')
+                rr = treecorr.NNCorrelation(min_sep=min_sep, max_sep=max_sep, bin_size=bin_size, sep_units='degrees')
+                rr.process(rand_cat)
+
+                #Random-Data 
+                rd = treecorr.NNCorrelation(min_sep=min_sep, max_sep=max_sep, bin_size=bin_size, sep_units='degrees')
+                rd.process(rand_cat, cat)
+
+                #Data-Random
+                dr = treecorr.NNCorrelation(min_sep=min_sep, max_sep=max_sep, bin_size=bin_size, sep_units='degrees')
+                dr.process(cat, rand_cat)
+
+                dd.write(os.path.join(output_dir, fname), rr, dr, rd)
+                xi, varxi = dd.calculateXi(rr, dr, rd)
+                xi_rad = np.exp(dd.meanlogr)
+                xi_sig = np.sqrt(varxi) 
+
+                plt.errorbar(xi_rad, xi, xi_sig, marker='o', ls='', c=color) 
+                plt.plot(validation_data['theta'], validation_data['w'], c=color, label=label)
+            plt.legend(loc=0)
+            plt.xscale('log')
+            plt.yscale('log')
+            plt.xlabel(r'$\theta$ deg')
+            plt.ylabel(r'$\xi(\theta)$')
+            #plt.text(1e-2, 5e-3, 'Lines: Wang et al 2013')
+            #plt.text(1e-2, 2e-3, 'Points: Catalog')
+            plt.title(self.possible_observations[self.observation]['label'])
+            plt.savefig(os.path.join(output_dir, 'xi_magnitude.png'), bbox_inches='tight')
+            if self.TestParams['savepdf']:
+                plt.savefig(os.path.join(output_dir, 'xi_magnitude.pdf'), bbox_inches='tight')
+
+
+        if self.observation == 'Zehavi2011_rAbsMagSDSS':
```

Comment:
This part of the if statement is still very much similar to the previous if statement. Is it possible to combine the identical parts between these two chunks?

Reply:
Let me know if you'd like some help on this one.

Reply:
@yymao I just added a function modify_figure() to modify the content and save the figure. I am not quite sure how to combine the catalog level searches for the two chunks. If you prefer changing the current lines please go ahead. Personally, I may not  prefer that.